src/app/clusters/bindings/PendingNotificationMap.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/binding-table.h>
+#include <lib/core/DataModelTypes.h>
+
+namespace chip {
+
+struct PendingNotificationEntry
+{
+public:
+    uint8_t mBindingEntryId;
+    void * mContext;
+};
+
+// The pool for all the pending comands.
+class PendingNotificationMap
+{
+public:
+    static constexpr uint8_t kMaxPendingNotifications = EMBER_BINDING_TABLE_SIZE;
+
+    friend class Iterator;
+
+    class Iterator
+    {
+    public:
+        Iterator(PendingNotificationMap * map, int16_t index) : mMap(map), mIndex(index) {}
+
+        PendingNotificationEntry operator*()
+        {
+            return PendingNotificationEntry{ mMap->mPendingBindingEntries[mIndex], mMap->mPendingContexts[mIndex] };
+        }
+
+        Iterator operator++()
```

Comment:
This does not look right to me.

Concretely, say kMaxPendingNotifications == mNumEntries == 3, and mNextToOverride == 1.

mIndex starts at 1.  Then it becomes 2.  Then it becomes 3, tests true for == mNumEntries, becomes -1, and iteration ends. But we never iterated the thing at index 0.

Reply:
Since we won't have more entries than those in the binding table, `mNextToOverride` will be always zero. Will remove the member and the related logic.